src/firefly/java/edu/caltech/ipac/visualize/net/AtlasImageGetter.java

```diff
@@ -0,0 +1,99 @@
+ * @version $Id: DssImageGetter.java,v 1.9 2012/08/21 21:30:41 roby Exp $
+ */
+public class AtlasImageGetter {
+
+
+    public static File lowlevelGetIbe2Image(BaseIrsaParams params)
+            throws FailedRequestException,
+            IOException {
+
+        try {
+            String sizeStr = null;
+            IbeDataSource ibeSource = null;
+            Map<String, String> queryMap = new HashMap<String, String>(11);
+            if (params instanceof AtlasImageParams) {
+                AtlasImageParams atlasParams = (AtlasImageParams) params;
+                sizeStr= atlasParams.getSize()+"";
+                ibeSource = new AtlasIbeDataSource();
+
+                Map<String, String> m = new HashMap<String, String>(1);
+                m.put(AtlasIbeDataSource.BAND_KEY, atlasParams.getBand());
+                m.put(AtlasIbeDataSource.INSTRUMENT_KEY, atlasParams.getInstrument());
+                if(atlasParams.getDs()!=null){ // using AtlasIbeSource defined datasets
+                    m.put(AtlasIbeDataSource.DS_KEY, atlasParams.getDs());
+                }else{
+                    m.put(AtlasIbeDataSource.DATASET_KEY, atlasParams.getSchema());
+                    m.put(AtlasIbeDataSource.TABLE_KEY, atlasParams.getTable());
+                    //Extra filter for querying images for now - see IRSA-
+                    queryMap.put(AtlasIbeDataSource.XTRA_KEY, atlasParams.getXtraFilter());
+                    queryMap.put(AtlasIbeDataSource.BAND_KEY, atlasParams.getBand());
+                    queryMap.put(AtlasIbeDataSource.INSTRUMENT_KEY, atlasParams.getInstrument());
+                }
+                ibeSource.initialize(m);
+
+            } else {
+                Assert.argTst(false, "unknown request type");
+            }
+            IBE ibe = new IBE(ibeSource);
+
+
+            File queryTbl = File.createTempFile("Ibe2Query-", ".tbl", CacheHelper.getDir());
+
+            IbeQueryParam queryParam = ibeSource.makeQueryParam(queryMap);
+            queryParam.setPos(params.getRaJ2000String() + "," + params.getDecJ2000());
+            queryParam.setMcen(true);
+            //queryParam.setIntersect(IbeQueryParam.Intersect.CENTER);
+            ibe.query(queryTbl, queryParam);
+
+            DataGroup data = IpacTableReader.readIpacTable(queryTbl, "results");
+
+
+            if (data.values().size() == 1) {
+                DataObject row = data.get(0);
+                Map<String, String> dataMap = IpacTableUtil.asMap(row);
+                IbeDataParam dataParam = ibeSource.makeDataParam(dataMap);
+
+                dataParam.setCutout(true, params.getRaJ2000String() + "," + params.getDecJ2000String(), sizeStr);
+                dataParam.setDoZip(true);
+                FileInfo result = ibe.getData(dataParam, null);
+                return new File(result.getInternalFilename());
+            } else {
+                throw new FailedRequestException("Area not covered, images #"+data.values().size());
```

Comment:
can you test this exception. This message will go area the way to the user.  Should it just say `Area not covered`?

Reply:
the problem is that the data can come back with more than 1 row and that doesn't mean that the area is not covered, it means that we don't know which one to pick among the result of this search. If you have a better message, I'm open to change it also. I've found the problem while implementing the IB2 API, depending on the search you can get more than one answer until we get the 'principal' column which will determine the main image to be displayed.